### PR TITLE
Add DLQ APIs in history service for history replication DLQ

### PR DESCRIPTION
**Pull request by yux0**

### CODE REVIEW

File: thrift/history.thrift [hunk 1]
```diff
@@ -857,4 +857,39 @@ service HistoryService {
...
+
+  /**
+  * MergeDLQMessages merges messages from DLQ
+  **/
+  replicator.MergeDLQMessagesResponse MergeDLQMessages(1: replicator.MergeDLQMessagesRequest request)
+    throws (
+      1: shared.BadRequestError badRequestError,
+      2: shared.InternalServiceError internalServiceError,
+      3: shared.ServiceBusyError serviceBusyError,
+      4: shared.EntityNotExistsError entityNotExistError,
```
Comment (yycptt) [inline thrift/history.thrift]: `ShardOwnerShipLostError`?

Comment (yux0) [reply thrift/history.thrift]: Good catch. Updated

File: thrift/history.thrift [hunk 2]
```diff
@@ -857,4 +857,40 @@ service HistoryService {
       6: ShardOwnershipLostError shardOwnershipLostError,
       7: shared.EntityNotExistsError entityNotExistError,
     )
+
+  /**
+  * ReadDLQMessages returns messages from DLQ
+  **/
+  replicator.ReadDLQMessagesResponse ReadDLQMessages(1: replicator.ReadDLQMessagesRequest request)
```
Comment (venkat1109) [inline thrift/history.thrift]: Does it make sense to name these ReadReplicatorDLQMessages ? If there is any other type of DLQ in the future, then the naming becomes confusing.

Comment (yux0) [reply thrift/history.thrift]: My plan is to create a generic API for DLQ. The request contains a type to support multiple types.

Comment (venkat1109) [reply thrift/history.thrift]: Makes sense. In that case is the return type 'replicator.ReadDLQMessagesResponse' also generic ?

Comment (yux0) [reply thrift/history.thrift]: Yes. It also includes the DLQ type

**Pull request merged**